pkg/operator/staticpod/prune/cmd.go

```diff
@@ -128,6 +128,12 @@ func (o *PruneOptions) Run() error {
 		return nil
 	}
 
+	// If the cert dir does not exist, do nothing.
+	// The dir will get eventually created by an installer pod.
+	if _, err := os.Stat(path.Join(o.ResourceDir, o.CertDir)); os.IsNotExist(err) {
```

Comment:
I think that `filepath.Walk` doesn't panic, check out - https://go.dev/play/p/FH3D3rPcQO7

Do you have a stacktrace?

Reply:
It's not the filepath.Walk which panics. It is https://github.com/openshift/library-go/blob/master/pkg/operator/staticpod/prune/cmd.go#L48. If unchecked the Run() returns an error for non-existing directory. Whereas, it's better to return nil when the directory does not exist.